plumbing/format/packfile/decoder_test.go

```diff
@@ -65,15 +69,35 @@ func (nsr nonSeekableReader) Read(b []byte) (int, error) {
 	return nsr.r.Read(b)
 }
 
-func (s *ReaderSuite) TestDecodeNoSeekable(c *C) {
+func (s *ReaderSuite) TestDecodeNoSeekableWithTxStorer(c *C) {
 	fixtures.Basic().ByTag("packfile").Test(c, func(f *fixtures.Fixture) {
 		reader := nonSeekableReader{
 			r: f.Packfile(),
 		}
 
-		scanner := NewScanner(reader)
+		scanner := packfile.NewScanner(reader)
 		storage := memory.NewStorage()
-		d, err := NewDecoder(scanner, storage)
+		d, err := packfile.NewDecoder(scanner, storage)
+		c.Assert(err, IsNil)
+		defer d.Close()
+
+		ch, err := d.Decode()
+		c.Assert(err, IsNil)
+		c.Assert(ch, Equals, f.PackfileHash)
+
+		assertObjects(c, storage, expectedHashes)
+	})
+}
+
+func (s *ReaderSuite) TestDecodeNoSeekableWithNoTxStorer(c *C) {
+	fixtures.Basic().ByTag("packfile").Test(c, func(f *fixtures.Fixture) {
+		reader := nonSeekableReader{
+			r: f.Packfile(),
+		}
+
+		scanner := packfile.NewScanner(reader)
+		storage, _ := filesystem.NewStorage(fs.New())
```

Comment:
How do you know this storage is not a TxStorer? It should be explicitly asserted in the test. Also the opposite should be asserted in the previous test.